tools/minify/cmd/configure.go

```diff
@@ -0,0 +1,117 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf/tools/minify/common"
+
+	"github.com/spf13/cobra"
+)
+
+var configureParams = struct {
+	AllYes bool
+	DryRun bool
+}{}
+
+func init() {
+	rootCmd.AddCommand(configureCmd)
+	configureCmd.Flags().BoolVarP(&configureParams.AllYes, "all-yes", "a", false, "Select all plugins")
+	configureCmd.Flags().BoolVarP(&configureParams.DryRun, "dry-run", "d", false, "Do all steps but do not generate the code files")
+}
+
+const configureLongDescription = `
+Applies the build-configuration by creating the all.go files in
+preparation of the actual Telegraf build.
+`
+
+var configureCmd = &cobra.Command{
+	Use:   "configure [flags]",
+	Short: "Configure Telegraf to prepare build",
+	Long:  configureLongDescription,
+	RunE:  configureWrapper,
+}
+
+func configureWrapper(cmd *cobra.Command, args []string) error {
+	return configure()
+}
+
+func configure() error {
+	// Check configuration options
+	if configureParams.AllYes && (len(allParams.ConfigFiles) != 0 || len(allParams.ConfigDirs) != 0) {
+		return errors.New("cannot use --all-yes and configuration files at the same time")
+	}
+
+	// Setup the configuration
+	var cfg *common.BuildConfig
+	if configureParams.AllYes {
+		// Include all plugins for --all-yes
+		cfg = common.NewConfiguration(true)
+	} else if len(allParams.ConfigFiles) > 0 || len(allParams.ConfigDirs) > 0 {
+		// Import the plugin list from Telegraf configuration files
+		log.Println("Importing configuration file(s)...")
+		cfg = common.NewConfiguration(false)
+		nfiles, err := cfg.ImportConfigurations(allParams.ConfigFiles, allParams.ConfigDirs)
+		if err != nil {
+			return fmt.Errorf("importing configuration(s) failed: %w", err)
+		}
+		if allParams.Verbose {
+			log.Printf("Found %d configuration files...", nfiles)
+		}
+	}
+
+	// Check if we do have a config
+	if cfg == nil {
+		return errors.New("no configuration found")
+	}
+
+	// Collect all available plugins
+	packages := common.PackageCollection{Repository: TelegrafRepository}
+	if err := packages.CollectAvailable(); err != nil {
+		return fmt.Errorf("collecting plugins failed: %w", err)
+	}
+
+	// Process the plugin list with the given config. This will
+	// only keep the plugins that adhere to the filtering criteria.
+	enabled, err := cfg.Filter(packages)
+	if err != nil {
+		return fmt.Errorf("filtering plugins failed: %w", err)
+	}
+	if allParams.Verbose {
+		enabled.Print()
+	}
+
+	// Extract the include paths
+	includes := enabled.ExtractIncludes()
+
+	// Write the import files
+	if !configureParams.DryRun {
+		if allParams.Verbose {
+			log.Println("Writing import files...")
+		}
+		if err := common.WriteImportFiles(includes); err != nil {
+			return fmt.Errorf("writing import files failed: %w", err)
+		}
+	}
+
```

Comment:
Can we please print what inputs, outputs, processors, aggregators, & parsers were enabled? That way we know for certain what minify enabled.

Similar to how we show what were loaded when we start up:

```
2022-07-28T14:07:45Z I! Loaded inputs: disk
2022-07-28T14:07:45Z I! Loaded aggregators: 
2022-07-28T14:07:45Z I! Loaded processors: 
2022-07-28T14:07:45Z I! Loaded outputs: file
```

Reply:
This can be done using `--verbose` as the README and help suggests... The list can be quite long and hide useful information... Do you want me to output this by default and reverse the `--verbose` flag?

Reply:
If someone has a problem with the tool it seems like we will always need to request the logs with `--verbose` to fully understand what is going on or where things failed. I'm inclined to say the `--verbose` logs should be what is printed by default.

Reply:
Ok, so I will invert `--verbose` to `--quiet` then...